lib/mssql.js

```diff
@@ -414,20 +430,26 @@ MsSQL.prototype.buildInsert = function (model, data) {
   var params = [];
   var props = this._models[model].properties;
   var modelPKID = this.idName(model);
-  if (props[modelPKID].generated !== false) {
+
+  var isIdentity = (props[modelPKID].type === Number &&
+    props[modelPKID].generated !== false);
+
+  if (isIdentity && data[modelPKID] == null) {
     //remove the pkid column if it's in the data, since we're going to insert a
     // new record, not update an existing one.
     delete data[modelPKID];
     //delete the hardcoded id property that jugglindb automatically creates
-    delete data.id;
+    // delete data.id;
   }
-  Object.keys(data).forEach(function (key) {
```

Comment:
Make sense, its actually much faster according to http://jsperf.com/foreach-vs-loop/77